### PR TITLE
[TypeSpec-Clean] Update for Azure.AI.TextTranslation

### PR DESCRIPTION
**Pull request by markcowl**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/translation/Azure.AI.TextTranslation/tspconfig.yaml [hunk 1]
```diff
@@ -13,10 +13,10 @@ emit:
   - "@azure-tools/typespec-autorest"
 options:
   "@azure-tools/typespec-autorest":
-    azure-resource-provider-folder: "../data-plane/Azure.AI.TextTranslation"
+    azure-resource-provider-folder: "data-plane"
     emitter-output-dir: "{project-root}/.."
     examples-directory: "./examples"
-    output-file: openapi.json
+    output-file: "{azure-resource-provider-folder}/Azure.AI.TextTranslation/stable/v3.0/openapi.json"
```
Comment (mikeharder) [inline specification/translation/Azure.AI.TextTranslation/tspconfig.yaml:19]:
```suggestion
    output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/openapi.json"
```

We should be able to use this config, if we also change the namespace from `TextTranslation` to `Azure.AI.TextTranslation`.

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: specification/translation/Azure.AI.TextTranslation/tspconfig.yaml [hunk 2]
```diff
@@ -13,10 +13,10 @@ emit:
   - "@azure-tools/typespec-autorest"
 options:
   "@azure-tools/typespec-autorest":
-    azure-resource-provider-folder: "../data-plane/Azure.AI.TextTranslation"
+    azure-resource-provider-folder: "data-plane"
```
Comment (mikeharder) [inline specification/translation/Azure.AI.TextTranslation/tspconfig.yaml:16]:
```suggestion
    azure-resource-provider-folder: "./data-plane"
```

NIT: Change to `./data-plane` to match sample.

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: specification/translation/Azure.AI.TextTranslation/tspconfig.yaml (hunk 1 above)
Comment (markcowl) [reply specification/translation/Azure.AI.TextTranslation/tspconfig.yaml:19]: @mikeharder I think that change is very likely to break SDKs generated from this.  Also, we cannot recreate the 'v3.0' from the interpolated variable, the versions tring will be '3.0'  personally, I think the version string is fine, but putting the literal string for the service in t config seems like a lesser evil than changing the namespace

Comment (mikeharder) [reply specification/translation/Azure.AI.TextTranslation/tspconfig.yaml:19]: OK, let's keep these custom `output-file` strings in the services that need them for now, and we can revisit this later in a push to make the specs as consistent as possible.

**Review by mikeharder — APPROVED** [review]

**Pull request merged**